Mobile_Detect.php

```diff
@@ -1317,30 +1318,29 @@ public function version($propertyName, $type = self::VERSION_TYPE_STRING)
         $properties = self::getProperties();
 
         // Check if the property exists in the properties array.
-        if (true === isset($properties[$propertyName])) {
+        if (!isset($properties[$propertyName])) {
+        	return false;
+        }
 
-            // Prepare the pattern to be matched.
-            // Make sure we always deal with an array (string is converted).
-            $properties[$propertyName] = (array) $properties[$propertyName];
+        // Prepare the pattern to be matched.
+        // Make sure we always deal with an array (string is converted).
+        $properties[$propertyName] = (array) $properties[$propertyName];
 
-            foreach ($properties[$propertyName] as $propertyMatchString) {
+        foreach ($properties[$propertyName] as $propertyMatchString) {
 
-                $propertyPattern = str_replace('[VER]', self::VER, $propertyMatchString);
+            $propertyPattern = str_replace('[VER]', self::VER, $propertyMatchString);
 
-                // Identify and extract the version.
-                preg_match(sprintf('#%s#is', $propertyPattern), $this->userAgent, $match);
+            // Identify and extract the version.
+            preg_match(sprintf('#%s#is', $propertyPattern), $this->userAgent, $match);
 
-                if (false === empty($match[1])) {
-                    $version = ($type == self::VERSION_TYPE_FLOAT ? $this->prepareVersionNo($match[1]) : $match[1]);
-
-                    return $version;
-                }
+            if (!empty($match[1])) {
+                $version = ($type == self::VERSION_TYPE_FLOAT ? $this->prepareVersionNo($match[1]) : $match[1]);
 
+                return $version;
             }
 
         }
 
-        return false;
```

Comment:
In this improvement, you are assuming that the `foreach` will always return something. But what could happen `$match[1]` (line 1336) is always empty? I would keep it as last chance.

Reply:
I agree.